store/src/java/com/zimbra/cs/service/FileUploadServlet.java

```diff
@@ -712,17 +714,35 @@ public void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOEx
         List<FileItem> items = new ArrayList<FileItem>(1);
         items.add(fi);
         Upload up = new Upload(acct.getId(), fi, filename);
-
-        if (filename.endsWith(".har")) {
-            File file = ((DiskFileItem) fi).getStoreLocation();
-            try {
-                Tika tika = new Tika();
-                String mimeType = tika.detect(file);
-                if (mimeType != null) {
-                    up.contentType = mimeType;
-                }
-            } catch (IOException e) {
-                mLog.warn("Failed to detect file content type");
+        File file = ((DiskFileItem) fi).getStoreLocation();
+        Tika tika = new Tika();
+        String mimeType = null;
+        try {
+            mimeType = tika.detect(file);
+        } catch (Exception e) {
+            mLog.warn("Error detecting mime type" + e.getMessage());
+        }
+        if (filename.endsWith(".har") && mimeType != null) {
+            up.contentType = mimeType;
+        }
+        if (mimeType == null) {
+            mimeType = up.contentType;
+        }
+        String contentTypeBlacklist = LC.zimbra_file_content_type_blacklist.value();
+        List<String> blacklist = new ArrayList<String>();
+        if (!StringUtil.isNullOrEmpty(contentTypeBlacklist)) {
+            blacklist.addAll(Arrays.asList(contentTypeBlacklist.trim().split(",")));
+        }
+        for (String blacklistedContentType : blacklist) {
```

Comment:
we can simplify this using streams.
 if (blackist.stream().anyMatch(ctype -> ctype.equals(blacklistedContentType)) {
           
        }